src/main/java/com/github/devotedmc/hiddenore/HiddenOre.java

```diff
@@ -27,43 +23,42 @@
 
 	private static BreakTracking tracking;
 	private BukkitTask trackingSave;
-	
+
 	private static BlockBreakListener breakHandler;
 	private static ExploitListener exploitHandler;
 	private static List<WorldGenerationListener> worldGen;
 
 	@Override
 	public void onEnable() {
 		plugin = this;
-		
+
 		saveDefaultConfig();
 		reloadConfig();
 		Config.loadConfig();
-		
+
 		tracking = new BreakTracking();
 		tracking.load();
 		trackingSave = Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, new Runnable() {
 			public void run() {
 				tracking.save();
 			}
 		}, Config.trackSave, Config.trackSave);
-		
+
 		exploitHandler = new ExploitListener(plugin);
 		this.getServer().getPluginManager().registerEvents(exploitHandler, this);
 
 		breakHandler = new BlockBreakListener(plugin);
 		this.getServer().getPluginManager().registerEvents(breakHandler, this);
-				
+
 		commandHandler = new CommandHandler(this);
 		this.getCommand("hiddenore").setExecutor(commandHandler);
-		
+
 		worldGen = new ArrayList<WorldGenerationListener>();
-		
-		ConfigurationSection worldGenConfig = Config.instance.getWorldGenerations();
-		if (worldGenConfig != null) {
-			for (String key : worldGenConfig.getKeys(false)) {
-				this.getLogger().log(Level.INFO, "Registered Ore Generation Suppression Listener for World {0}", key);
-				WorldGenerationListener list = new WorldGenerationListener(worldGenConfig.getConfigurationSection(key));
+		for (World world : getServer().getWorlds()) {
+			ConfigurationSection worldGenConfig = Config.instance.getWorldGenerations(world.getName());
+			if (worldGenConfig != null) {
+				this.getLogger().log(Level.INFO, "Registered Ore Generation Suppression Listener for World {0}", world.getName());
+				WorldGenerationListener list = new WorldGenerationListener(worldGenConfig.getConfigurationSection("this_world"));
```

Comment:
I'd recommend eliminating the spurious "hardcoded" `this_world` segment. Not needed. Just pass `worldGenConfig` directly to WorldGenerationListener constructor. I'll leave a note in the example config remarking the corresponding change.